kong/plugins/key-auth/handler.lua

```diff
@@ -101,8 +137,24 @@ local function do_authentication(conf)
         clear_header(name)
 
         if conf.key_in_body then
-          body_data[name] = nil
-          ngx_req_set_body_data(ngx_encode_args(body_data))
+          if not hide_body_credentials[req_mime] then
+            -- the request was indeed well formed but could not be processed
+            -- the status '422' might be a good candidate here, but it's part
+            -- of the WebDAV extension, so it doesn't seem appropriate here
+            -- and a 5xx status seems inappropriate as well- the server (plugin)
+            -- configuration is not incorrect. it's up to the client to present
+            -- the appropriate body encoding, given the server configuration
+            -- this places an onus of responsibility on the server operator to
+            -- properly document the acceptable body encodings when
+            -- 'hide_credentials' and 'key_in_body' are both set
+            return false, { status = 400, message = "Cannot process request body" }
```

Comment:
Should we catch this condition earlier? Like:
```lua
body_data, err, raw_body, req_mime = public_tools.get_body_info()

if err or not hide_body_credentials[req_mime] then
  return false, { status = 400, message = "Cannot process request body" }
end
```

Ideally, the public API exposes a `can_parse()` utility to prevent reading the request body if we already know we cannot decode it, but it would already be a slight improvement?

Reply:
I'm not sure this approach is valid. Just because we cannot parse the body to search for the key does mean mean we should immediately fail. If the request contains an unparseable body but the auth key is in a header or query param, we don't need to mangle the body, and thus I don't believe we fail to process the request.

Reply:
True, let's not do that!